webapp/src/notifications/newOfferAvailable.ts

```diff
@@ -14,17 +14,48 @@ export async function sendNewOfferAvailable() {
   try {
     const payload = await getPayloadClient({ seed: false });
 
-    const todayOffers = await payload.find({
+    const currentDate = new Date();
+
+    let todayOffers = await payload.find({
       collection: "offers",
       pagination: false,
       depth: 1,
       where: {
-        createdAt: {
-          greater_than_equal: new Date().toISOString().split("T")[0],
+        validityFrom: {
+          greater_than_equal: `${
+            currentDate.toISOString().split("T")[0]
+          }T00:00:00`,
+          less_than_equal: `${
+            currentDate.toISOString().split("T")[0]
+          }T23:59:59`,
         },
       },
     });
 
+    await Promise.all(
+      todayOffers.docs.map(async (offer) => {
+        const offerCoupons = await payload.find({
+          collection: "coupons",
+          limit: 1,
+          where: {
+            offer: {
+              equals: offer.id,
+            },
+            user: {
+              exists: false,
+            },
+            used: {
+              equals: false,
+            },
+          },
+        });
+
+        if (offerCoupons.docs.length === 0) {
+          todayOffers.docs = todayOffers.docs.filter((o) => o.id !== offer.id);
```

Comment:
pas fan de la modification d'un tableau en cours d'itération, par principe

du coup tu peux peut être return le boolean dans ta promesse et ensuite traiter les resultats comme ça : 

```
await Promise.all(
  todayOffers.docs.map(async (offer) => {
    const offerCoupons = await payload.find({
      collection: "coupons",
      limit: 1,
      where: {
        offer: {
          equals: offer.id,
        },
        user: {
          exists: false,
        },
        used: {
          equals: false,
        },
      },
    });

    return offerCoupons.docs.length === 0;
  })
).then((results) => {
  todayOffers.docs = todayOffers.docs.filter((_, index) => results[index]);
});

```